SpiceSharp/Simulations/IntegrationMethods/Spice/Gear/Gear.cs

```diff
@@ -0,0 +1,248 @@
+            }
+        }
+
+        /// <summary>
+        /// Predict a new solution based on the previous ones
+        /// </summary>
+        /// <param name="simulation">Time-based simulation</param>
+        protected override void Predict(TimeSimulation simulation)
+        {
+            if (simulation == null)
+                throw new ArgumentNullException(nameof(simulation));
+
+            // Use the previous solutions to predict a new one
+            for (var i = 0; i <= IntegrationStates[0].Solution.Length; i++)
+            {
+                Prediction[i] = 0.0;
+                for (var k = 0; k <= Order; k++)
+                {
+                    Prediction[i] += PredictionCoefficients[k] * IntegrationStates[k + 1].Solution[i];
+                }
+            }
+        }
+
+        /// <summary>
+        /// Truncate the timestep using nodes
+        /// </summary>
+        /// <param name="sender">Sender</param>
+        /// <param name="args">Arguments</param>
+        protected override void TruncateNodes(object sender, TruncateEvaluateEventArgs args)
+        {
+            // Get the state
+            var state = args.Simulation.RealState;
+            var timetmp = double.PositiveInfinity;
+            var nodes = args.Simulation.Nodes;
+
+            var delsum = 0.0;
+            for (var i = 0; i <= Order; i++)
+                delsum += IntegrationStates[i].Delta;
+
+            for (var i = 0; i < nodes.Count; i++)
+            {
+                var node = nodes[i];
+                var index = node.Index;
+                var tol = Math.Max(Math.Abs(state.Solution[index]), Math.Abs(Prediction[index])) * RelTol + AbsTol;
+                var diff = state.Solution[index] - Prediction[i];
+
+                if (!diff.Equals(0.0))
+                {
+                    var tmp = tol * TrTol * delsum / (diff * IntegrationStates[0].Delta);
+                    tmp = Math.Abs(tmp);
+                    switch (Order)
+                    {
+                        case 0: break;
+                        case 1:
+                            tmp = Math.Sqrt(tmp);
+                            break;
+                        default:
+                            tmp = Math.Exp(Math.Log(tmp) / (Order + 1));
+                            break;
+                    }
+
+                    tmp *= IntegrationStates[0].Delta;
+                    timetmp = Math.Min(timetmp, tmp);
+                }
+
+                args.Delta = timetmp;
+            }
+        }
+
+        /// <summary>
+        /// Compute the coefficients for Trapezoidal integration
+        /// </summary>
+        protected override void ComputeCoefficients()
+        {
+            var delta = IntegrationStates[0].Delta;
+            for (var i = 0; i < Coefficients.Length; i++)
+                Coefficients[i] = 0.0;
+            Coefficients[1] = -1.0 / delta;
+
+            // First, set up the matrix
+            double arg = 0, arg1;
```

Comment:
@svenboulanger I don't know but is arg1 a good variable name?

Reply:
This is copied from Spice 3f5's implementation for Gear integration. It seems to be just another temporary variable like `arg` so I'm not sure what a better naming would be.